java.lang.reflect.InaccessibleObjectException

Hi, I'm currently integrating Stripe with a Spring Boot app of mine. I'm following the documentation described [here](https://stripe.com/docs/payments/bancontact/accept-a-payment)

Based of this I have a method which returns the PaymentIntentCreateParams object
```
private PaymentIntentCreateParams stripeParams(double amount) { 
    return PaymentIntentCreateParams.builder()
                .setAmount((long) (amount * 100))
                .setCurrency("eur")
                .addPaymentMethodType("bancontact")
                .setPaymentMethodOptions(
                        PaymentIntentCreateParams.PaymentMethodOptions.builder()
                                .putExtraParam("bancontact[preferred_language]", "nl")
                                .build())
                .build();
    }
```


Which is then used when calling a certain page
```

@GetMapping("/paymentRedirect/{name}/{lastname}/{email}/{desc}")
    public String getRedirect(Model model, @PathVariable String email, @PathVariable String desc, @PathVariable String name, @PathVariable String lastname, HttpServletRequest request) {
        System.out.println("Ik wordt geredirect: " + email + " " + desc + " " + name + " " + lastname);
        HttpSession session = request.getSession();

        // Set your secret key. Remember to switch to your live secret key in production.
        // See your keys here: https://dashboard.stripe.com/account/apikeys
        Stripe.apiKey =

                "sk_test_51IRJ8lH41Sol4wsKeKoZ9RXaE171jfBbLDDcDbkPlCi1tCz6Ca8HXsCG2AzXIY6YDaiXuGZ0gsD1nnee7X6gZGiO00LSkoKXJy"
        ;
        double price = (double) session.getAttribute("orderTotal");
        System.out.println("price: "+price);
        PaymentIntentCreateParams params = stripeParams(price);
        try {
            PaymentIntent intent = PaymentIntent.create(params);
            model.addAttribute("orderDTO", new OrderDTO());
            model.addAttribute("client_secret", intent.getClientSecret());
            model.addAttribute("name", name);
            model.addAttribute("surname", lastname);
            model.addAttribute("email", email);
            model.addAttribute("desc", desc);
            System.out.println("ID is set");
            System.out.println("Dit is de intent zijn id: " + intent.getId());
            session.setAttribute("id", intent.getId());

            return "main_pages/redirect";
        } catch (StripeException e) {
            e.printStackTrace();
            return "main_pages/error";
        }
    }
```

But when the actual method is called, PaymentIntent.create(params); throws an error: `java.lang.reflect.InaccessibleObjectException:` Unable to make private java.net.Proxy() accessible: module java.base does not "opens java.net" to unnamed module @7334aada`

Currently running Java SE 16

I ran into the same issue:
`Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make private java.net.Proxy() accessible: module java.base does not "opens java.net" to unnamed module @7ff95560
	at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357) ~[?:?]
	at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297) ~[?:?]
	at java.lang.reflect.Constructor.checkCanSetAccessible(Constructor.java:188) ~[?:?]
	at java.lang.reflect.Constructor.setAccessible(Constructor.java:181) ~[?:?]
	at com.google.gson.internal.reflect.UnsafeReflectionAccessor.makeAccessible(UnsafeReflectionAccessor.java:44) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.ConstructorConstructor.newDefaultConstructor(ConstructorConstructor.java:103) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.ConstructorConstructor.get(ConstructorConstructor.java:85) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:101) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.getAdapter(Gson.java:458) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.getAdapter(Gson.java:458) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.getAdapter(Gson.java:458) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.getDelegateAdapter(Gson.java:541) ~[gson-2.8.6.jar:?]
	at com.stripe.model.PaymentSourceTypeAdapterFactory.create(PaymentSourceTypeAdapterFactory.java:31) ~[stripe-java-20.8.0.jar:20.8.0]
	at com.google.gson.Gson.getAdapter(Gson.java:458) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory.create(CollectionTypeAdapterFactory.java:53) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.getAdapter(Gson.java:458) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.getAdapter(Gson.java:458) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166) ~[gson-2.8.6.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.getAdapter(Gson.java:458) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:931) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[gson-2.8.6.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:817) ~[gson-2.8.6.jar:?]
	at com.stripe.net.LiveStripeResponseGetter.request(LiveStripeResponseGetter.java:66) ~[stripe-java-20.8.0.jar:20.8.0]
	at com.stripe.net.ApiResource.request(ApiResource.java:179) ~[stripe-java-20.8.0.jar:20.8.0]
	at com.stripe.net.ApiResource.request(ApiResource.java:169) ~[stripe-java-20.8.0.jar:20.8.0]
	at com.stripe.model.Customer.create(Customer.java:240) ~[stripe-java-20.8.0.jar:20.8.0]
	at com.stripe.model.Customer.create(Customer.java:233) ~[stripe-java-20.8.0.jar:20.8.0]`

--- a/src/main/java/com/stripe/model/StripeCollection.java
+++ b/src/main/java/com/stripe/model/StripeCollection.java
@@ -50,7 +50,7 @@ public abstract class StripeCollection<T extends HasId> extends StripeObject
 
   @Getter(onMethod_ = {@Override})
   @Setter(onMethod = @__({@Override}))
-  private RequestOptions requestOptions;
+  private transient RequestOptions requestOptions;
 
   @Getter(onMethod_ = {@Override})
   @Setter(onMethod = @__({@Override}))

--- a/src/main/java/com/stripe/net/ApiResourceTypeAdapterFactoryProvider.java
+++ b/src/main/java/com/stripe/net/ApiResourceTypeAdapterFactoryProvider.java
@@ -18,6 +18,7 @@ final class ApiResourceTypeAdapterFactoryProvider {
     factories.add(new BalanceTransactionSourceTypeAdapterFactory());
     factories.add(new ExternalAccountTypeAdapterFactory());
     factories.add(new PaymentSourceTypeAdapterFactory());
+    factories.add(new ReflectionCheckingTypeAdapterFactory());
   }
 
   public static List<TypeAdapterFactory> getAll() {

--- a/src/main/java/com/stripe/net/ReflectionCheckingTypeAdapterFactory.java
+++ b/src/main/java/com/stripe/net/ReflectionCheckingTypeAdapterFactory.java
@@ -1,0 +1,26 @@
+package com.stripe.net;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.internal.bind.ReflectiveTypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * {@link TypeAdapterFactory} that checks that we don't use {@link ReflectiveTypeAdapterFactory}
+ * accidentally for classes outside {@code com.stripe} packages. This usually happens when we forget
+ * to mark a field {@code transient}.
+ */
+class ReflectionCheckingTypeAdapterFactory implements TypeAdapterFactory {
+  @Override
+  public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+    if (!type.getType().getTypeName().startsWith("com.stripe.")) {
+      TypeAdapter<T> adapter = gson.getDelegateAdapter(this, type);
+      if (adapter instanceof ReflectiveTypeAdapterFactory.Adapter) {
+        throw new IllegalArgumentException(
+            "Refusing to create type reflection-based type adapter for external class: " + type);
+      }
+    }
+    return null;
+  }
+}

